src/shared_modules/indexer_connector/src/indexerConnectorSyncImpl.hpp

```diff
@@ -499,4 +496,14 @@ class IndexerConnectorSyncImpl final
             processBulk();
         }
     }
+
+    std::mutex& scopeLock()
+    {
+        return m_mutex;
+    }
```

Comment:
Exposing internal mutex by reference violates encapsulation and can lead to:
- Deadlocks from inconsistent lock ordering
- Race conditions from uncontrolled external locking
- Maintenance issues when internal locking strategy changes

```suggestion
class ScopeGuard
{
public:
    explicit ScopeGuard(std::mutex& m) : m_lock(m) {}
    ScopeGuard(const ScopeGuard&) = delete;
    ScopeGuard& operator=(const ScopeGuard&) = delete;
private:
    std::unique_lock<std::mutex> m_lock;
};

ScopeGuard scopeLock()
{
    return ScopeGuard{m_mutex};
}
```
or template function